argus/unit_tests/action_manager/test_windows.py

```diff
@@ -0,0 +1,789 @@
+# Copyright 2016 Cloudbase Solutions Srl
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import requests
+import unittest
+import urlparse
+
+try:
+    import unittest.mock as mock
+except ImportError:
+    import mock
+
+from argus.action_manager import windows as action_manager
+from argus import exceptions
+from argus.unit_tests import test_utils
+from argus import util
+
+
+class WindowsActionManagerTest(unittest.TestCase):
+    """Tests for windows action manager class."""
+
+    def setUp(self):
+        self._client = mock.MagicMock()
+        self._config = mock.MagicMock()
+        self._os_type = mock.sentinel
+
+        self._action_manager = action_manager.WindowsActionManager(
+            client=self._client, config=self._config, os_type=self._os_type)
+
+    def _test_wait_boot_completion_function(self, run_command_exc=None):
+        if run_command_exc:
+            self._client.run_command_until_condition = mock.Mock(
+                side_effect=run_command_exc)
+            with self.assertRaises(run_command_exc):
+                action_manager.wait_boot_completion(
+                    self._client, test_utils.USERNAME)
+        else:
+            self._client.run_command_until_condition = mock.Mock()
+            self.assertIsNone(
+                action_manager.wait_boot_completion(
+                    self._client, test_utils.USERNAME))
+
+    def test_wait_boot_completion_function_successful(self):
+        self._test_wait_boot_completion_function()
+
+    def test_wait_boot_completion_function_fail(self):
+        self._test_wait_boot_completion_function(
+            run_command_exc=exceptions.ArgusTimeoutError)
+
+    def test_download_successful(self):
+        self._action_manager.download(test_utils.URI, test_utils.LOCATION)
+
+        cmd = 'Invoke-WebRequest -Uri {} -OutFile "{}"'.format(
+            test_utils.URI, test_utils.LOCATION)
+        self._client.run_command_with_retry.assert_called_with(
+            cmd, count=util.RETRY_COUNT, delay=util.RETRY_DELAY,
+            command_type=util.POWERSHELL)
+
+    def test_download_exception(self):
+        self._client.run_command_with_retry.side_effect = \
```

Comment:
It is preferred to wrap long lines in parentheses and not a backslash for line continuation.